src/java.desktop/share/classes/javax/swing/plaf/synth/SynthLookAndFeel.java

```diff
@@ -665,6 +665,10 @@ public void initialize() {
         setStyleFactory(factory);
         KeyboardFocusManager.getCurrentKeyboardFocusManager().
             addPropertyChangeListener(_handler);
+        if (UIManager.getLookAndFeel().getName().contains("GTK")) {
```

Comment:
This doesn't seem like an ideal way of doing this.
I don't see any precedent for this approach in the Swing implementation.

We need some method that is more about the properties of a L&F rather than keying off name.

Please look around at LookAndFeel and UIDefaults (etc) for a place you could store a property that tells you what to do.
Or perhaps you this "altProcessor" would be the value you retrieve and if it is non-null you install it. 
Or perhaps this should be pushed down to the GTK L&F

I don't know what the right answer is, I just don't think this approach is it.

Reply:
> Or perhaps this should be pushed down to the GTK L&F

I tried to have this implementation in GTK L&F but there were some build issue I was facing. So, moved it to Synth L&F but that impacts Nimbus L&F as well. 

> This doesn't seem like an ideal way of doing this.
I don't see any precedent for this approach in the Swing implementation.
We need some method that is more about the properties of a L&F rather than keying off name.
Please look around at LookAndFeel and UIDefaults (etc) for a place you could store a property that tells you what to do.
Or perhaps you this "altProcessor" would be the value you retrieve and if it is non-null you install it.

Sure, I will check for the fix  other way around.

Reply:
@prrace Added the property to handle ALT key press specific to GTK L&F. Based on the property value the event handler for ALT key press is installed.